tasks/create-advisory/create-advisory.yaml

```diff
@@ -65,6 +65,15 @@ spec:
         advisoryData=$(jq -c "$(params.jsonKey)" $(workspaces.data.path)/$(params.dataPath))
         configMapName=$(jq -er '.sign.configMapName' $(workspaces.data.path)/$(params.dataPath))
 
+        # Validate type
+        if jq -e "$(params.jsonKey).type" $(workspaces.data.path)/$(params.dataPath) > /dev/null 2>&1; then
```

Comment:
Is the type optional? If it is, I don't see a default being set anywhere here. Would it be set in the internal request? Also, why not use `$advisoryData` instead of reading the file again? Also, I would suggest using `has`, e.g. `if jq -e 'has("type")' <<< $advisoryData` or `if [ $(jq 'has("type")' <<< $advisoryData) == false ]` which is what we have elsewhere.

Reply:
No, it is not optional
https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/check-data-keys/check-data-keys.yaml#L40
I don't think it makes sense to once again check that the key exists

Reply:
Set it as optional as it is [optional in Konflux-release-data](https://gitlab.cee.redhat.com/releng/konflux-release-data/-/blob/main/schemas/extra/releaseplanadmission.json?ref_type=heads#L24). I'll change it here.

Reply:
That is setting it as optional in the RPA. That is true. It doesn't have to be in the RPA. It can be in the Release or ReleasePlan. It is not optional in the resulting data file though

Reply:
Oh right, because the type can be defined in the RP or the Release right?

Reply:
Yep. In my original design I actually thought it would be in the Release as it would be set based on # of CVEs/Jira issues found